### PR TITLE
Docs/devcontainer

### PR DESCRIPTION
**Pull request by wolflu05**
I tried to write a first draft of the devcontainer docs. Please suggest things that need to be added too and also correct language if you think there could be something improved.

### CODE REVIEW

**Comment by SchrodingersGat** [comment]
@wolflu05 following your instructions through and I hit an error at the "open in container" step:

image

Do you see this error on your end?

**Review by matmair — APPROVED** [review]

**Comment by wolflu05** [comment]
> Do you see this error on your end?

No, haven't got that error. What system are you running on?

**Comment by SchrodingersGat** [comment]
> > Do you see this error on your end?
> 
> No, haven't got that error. What system are you running on?

Running on windows 10. My normal dev setup is under docker but I've not used Dev containers before

**Comment by wolflu05** [comment]
May you can try a fresh clone of the repo somewhere else, this must have something todo with permissions. Maybe your files have different permissions that work with the user in the other dev image but not with the vscode user used in the devcontainer image.

**Comment by SchrodingersGat** [comment]
Ah ok worked it out, I had cloned into a directory with a different name - was not called `InvenTree`. There are some hard-coded paths in the files in `.devcontainer` which cause a ruckus if the directory is named differently.

Either we can:

1. Address this and  not hard-code the paths
2. Add a clear note to the docs about this limitation

Also, probably should add a note that if anything goes wrong during the setup process, the user should delete the `dev` directory before starting again.

**Comment by wolflu05** [comment]
Good that you figured that out. Maybe we can utilize these [variables](https://containers.dev/implementors/json_reference/#variables-in-devcontainerjson) to get it generic?  I think `${containerWorkspaceFolder}` looks promising?

**Comment by wolflu05** [comment]
@SchrodingersGat can you verify that this works now with your different named repo clone too?

**Comment by matmair** [comment]
@wolflu05 if there is a clear warning (we support warning blocks with mkdocs) it should be fine. It should however not be directly a part of the getting started IMO.

**Comment by SchrodingersGat** [comment]
I had to make a couple of further small changes to get it to work - see attached patch file.

[devcontainer.patch](https://github.com/inventree/InvenTree/files/11430988/devcontainer.patch)

**Comment by wolflu05** [comment]
@SchrodingersGat yeah thanks, I totally forgot about that other two files. Applied your patch, just added a small view changes and removed the change in `.pre-commit-config.yaml`. Also added a quick not about the `dev` folder to the FAQ. 

Also discovered that I set up something additional in my plugins folder vscode config for better auto complete with InvenTree core code in plugins. This is useful if you're dealing with some Mixins imported from InvenTree core.

File: docs/docs/start/devcontainer.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+   "mounts": [
+     "source=/path/to/your/local/inventree-plugin,target=/workspaces/inventree-plugin,type=bind,consistency=cached"
+   ],
+   ```
+2. Add `/workspaces/inventree-plugin` to your vscode workspace folders by click on `File > Add folder to workspace…`.
+3. Install your plugin as pip editable install by executing the following command in the venv.
+   ```bash
+   pip install -e /workspaces/inventree-plugin
+   ```
+4. Add InvenTree core code to pylance intelicense path by adding the following file to your plugin repo `.vscode/settings.json` (Your path can be different depending on your setup):
```
Comment (matmair) [inline docs/docs/start/devcontainer.md:72]: Suggestion: I think this is a typo?
```suggestion
4. Add InvenTree core code to Pylance IntelliSense path by adding the following file to your plugin repo `.vscode/settings.json` (Your path can be different depending on your setup):
```

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -65,13 +65,15 @@ nav:
     - Configuration: start/config.md
     - Docker:
         - Introduction: start/docker.md
-        - Development: start/docker_dev.md
```
Comment (matmair) [inline docs/mkdocs.yml:68]: I think the setup instructions should stay here and be linked in an overview dev doc.

File: docs/mkdocs.yml [hunk 3]
```diff
@@ -65,13 +65,15 @@ nav:
...
         - Production: start/docker_prod.md
     - Bare Metal:
         - Introduction: start/install.md
         - Installer: start/installer.md
-        - Development: start/development.md
```
Comment (matmair) [inline docs/mkdocs.yml:73]: See above

**Review by matmair — APPROVED** [review]

File: docs/mkdocs.yml (hunk 2 above)
Comment (wolflu05) [reply docs/mkdocs.yml:68]: And where should I put the devcontainer doc? Directly at the same level of docker?

Comment (matmair) [reply docs/mkdocs.yml:68]: I think it is good where it is. We are dragging around the idea of some better dev docs for a while now. This PR can be the start and I will try to formulate something for a general getting started from this: https://github.com/inventree/InvenTree/issues/4793#issuecomment-1540758748

Comment (wolflu05) [reply docs/mkdocs.yml:68]: Sorry, don’t get how I should do it, maybe you can push it like you want?

Comment (matmair) [reply docs/mkdocs.yml:68]: I can if that is ok with you; I do not like pushing to other ppl. branches, feels like overstepping

Comment (wolflu05) [reply docs/mkdocs.yml:68]: Yeah, absolutely fine.

**Comment by SchrodingersGat** [comment]
@wolflu05 LGTM now - please respond to the change by @matmair then I'm happy to merge in :)

**Comment by matmair** [comment]
I pushed the suggested changes as agreed @wolflu05 ; should be fine for a merge @SchrodingersGat

**Comment by SchrodingersGat** [comment]
@wolflu05 thanks for the contributions :)

**Comment by wolflu05** [comment]
> I pushed the suggested changes as agreed @wolflu05 ; should be fine for a merge @SchrodingersGat 

That was easy 😀 - thought you want also a second page added for quick overview.

**Pull request merged**